packages/wrangler/src/__tests__/secret.test.ts

```diff
@@ -194,21 +229,27 @@ describe("wrangler secret", () => {
 
 				// User counter to pass different secrets to the request mock
 				let counter = 0;
-				setMockResponse(
-					`/accounts/:accountId/workers/scripts/:scriptName/secrets`,
-					"PUT",
-					([_url, accountId]) => {
-						expect(accountId).toEqual("some-account-id");
-						counter++;
-
-						if (counter % 2 === 0) {
-							return { name: `secret-name-${counter}`, type: "secret_text" };
-						} else {
-							return Promise.reject(
-								new Error(`Failed to create secret ${counter}`)
-							);
+				msw.use(
+					rest.put(
+						`*/accounts/:accountId/workers/scripts/:scriptName/secrets`,
+						(req, res, ctx) => {
+							expect(req.params.accountId).toEqual("some-account-id");
+							counter++;
+
+							if (counter % 2 === 0) {
+								return res(
+									ctx.json(
+										createFetchResult({
+											name: `secret-name-${counter}`,
+											type: "secret_text",
+										})
+									)
+								);
+							} else {
+								return res.networkError(`Failed to create secret ${counter}`);
```

Comment:
I am not sure a network error is the correct failure here?
Isn't it supposed to be more like a 500 error?

Reply:
Previously they were just throwing a `Promise.reject(new Error)` would that be more align with a network error or a bad API response, it was unclear to me?

Reply:
Did a pair programming session with Pete determined that the `.networkError` had the same results as the returning a `500 response` however networkError returns faster in MSW and allows to send a message for testing purposes. 

@rozenmd The `fetch error` with the URL is how it would be sent in a real world situation, due to a network failure or a 500